src/main/java/com/poiji/bind/Poiji.java

```diff
@@ -359,11 +360,15 @@ private static HSSFPropertyStream deserializerPropertyStream(PoijiExcelType exce
      * @see Poiji#fromExcel(Sheet, Class, PoijiOptions)
      * @see Poiji#fromExcel(Sheet, Class)
      */
-    public static <T> void fromExcel(final Sheet sheet,
+    public static <T> void fromExcel(Sheet sheet,
                                      final Class<T> type,
                                      final PoijiOptions options,
                                      final Consumer<? super T> consumer) {
         Objects.requireNonNull(sheet);
+        if(options.getColumnsToKeep()!=null)
+            sheet = SheetColumns.sheetAdaptor(sheet,options);
+        options.setSkip(options.skip()-options.getHeaderStart());
```

Comment:
why we need to use them here?
```
options.setSkip(options.skip()-options.getHeaderStart());
options.setHeaderStart(0);
```

Reply:
skip was itself increasing itself to skip+headerstart.
in new virtual sheet the headers are at 0
Request support from more experienced..